CONTRIBUTING.md

```diff
@@ -0,0 +1,64 @@
+
+We take our open source community seriously and hold ourselves and other contributors to high standards of communication.
+By participating and contributing to this project, you agree to uphold our [Code of conduct](https://www.contributor-covenant.org/version/2/1/code_of_conduct/).
+
+## Report a bug or suggest a feature
+
+1. Search whether your bug or feature was already reported. You can do so on [the GitHub issue tracker of nixos/nix](https://github.com/NixOS/nix/issues).
+
+2. If you were not able to find the bug or feature [open a new issue](https://github.com/NixOS/nix/issues/new/choose)
+
+3. The issue template will guide you in what should be in an issue.
+   The more complete the information, the more likely it can be found by others and the more useful it is in the future.
+   Try to make sure reported bugs can be reproduced by other contributors.
+
+4. Once submitted, do not expect issues to be picked up or solved right away.
+   The only way to ensure this, is to work on the issue yourself. See [Making changes to Nix](#making-changes-to-nix).
+
+## Report a security vulnerability
+
+Check out the [security policy](https://github.com/NixOS/nix/security/policy).
+
+## Making changes to Nix
+
+1. Search for already existing pull requests that might cover the contribution you are about to make.
+   There are many open pull requests that might already do what you intent to work on.
+
+2. Search for related issues that cover what you're going to work on. It could help to mention there that you will work on the issue.
+
+3. Check [the manual](https://nixos.org/manual/nix/stable/contributing/hacking.html) for information on building Nix and running its tests.
+   
+   For contributions on the CLI, please check the manual page on [CLI guidelines](https://nixos.org/manual/nix/stable/contributing/cli-guideline.html).
+   
+4. Make your changes!
+
+5. [Create a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) for your changes
+   * [Mark the pull request as draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) if you're not done with the changes.
+   * Make sure to have [a clean history of commits on your branch by using rebase](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request).
+   * Link related issues in your pull request to inform interested parties and future contributors about your change.
+
+6. Do not expect your pull request to be reviewed immediately.
+   The core Nix developers are overwhelmed with notifications and reviewing other pull requests.
+
+7. Ask other contributors for feedback and to shepherd your pull request into shape.
+   You can do so by opening a new topic on [Discourse in the Nix Development category](https://discourse.nixos.org/c/dev/nix/50).
```

Comment:
```suggestion
7. If you need additional feedback or help to getting pull request into shape, ask other contributors using [@mentions](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#mentioning-people-and-teams).
```

Almost no one working on the code will look at Discourse, so it probably won't help. Also I wouldn't want that category to be flooded with "please review my PR" topics. Additionally, the discourse bot will make a comment every time someone links a PR on Discourse, and while often useful, this instruction would produce lots of unnecessary visual noise.

Reply:
I'm not sure about this one. Having it be posted on discourse was what helped me get feedback. I do agree it can create visual noise of used too often.

Mentioning a contributor is an ok mechanism, but which one do you pick? A new contributor will likely just check who worked on a file last time and mention that person. There is a good chance that is a core contributor, which won't help to mention.

Is there a method/process to find an existing contributor that isn't on the core team?

For nixpkgs there is also https://discourse.nixos.org/t/prs-ready-for-review/3032/1380, but i wouldn't want to disturb that process with Nix PRs. And, it isn't a very nice process 😅

Reply:
These are three separate issues:
1. asking questions about the code and development
   
   This is what works best on Discourse, and this is what the Nix Category is for regarding this repo, and this is how you came here. It works, apparently, and I'd encourage it.

2. finding the right people to ask directly

   No obvious silver bullet, but git blame sounds like a good candidate. Yes, that's likely to turn out maintainers, but in the longer run this will diversify and already does.

3. asking for review and help on a given PR

   This should absolutely be done as @mentions, in place

Reply:
Thanks, I've updated and now also formatted the documentation. I'm hoping this is enough information for contributors to go on. It should already help a bit.

Let me know if you have any other suggestions or what I can do to get this reviewed further and/or merged.